src/split_at.rs

```diff
@@ -335,4 +827,32 @@ mod tests {
         test_split_at::<1, 17>();
         test_split_at::<2, 18>();
     }
+
+    #[cfg(feature = "derive")]
+    #[test]
+    #[allow(clippy::as_conversions)]
+    fn test_split_at_overlapping() {
```

Comment:
Also add a test (probably in a different function than this one) for `via_runtime_check` failing?

Reply:
What do you mean? The test here is that `via_runtime_checks` succeeds for odd indices and fails for even ones.

Reply:
Oops, I just misread this test.